smack-experimental/src/main/java/org/jivesoftware/smackx/reference/element/ReferenceElement.java

```diff
@@ -0,0 +1,141 @@
+/**
+ *
+ * Copyright 2018 Paul Schaub
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jivesoftware.smackx.reference.element;
+
+import org.jivesoftware.smack.packet.ExtensionElement;
+import org.jivesoftware.smack.util.XmlStringBuilder;
+import org.jivesoftware.smackx.reference.ReferenceManager;
+
+public class ReferenceElement implements ExtensionElement {
+
+    public static final String ELEMENT = "reference";
+    public static final String ATTR_BEGIN = "begin";
+    public static final String ATTR_END = "end";
+    public static final String ATTR_TYPE = "type";
+    public static final String ATTR_ANCHOR = "anchor";
+    public static final String ATTR_URI = "uri";
+
+    public enum Type {
+        mention,
+        data
+    }
+
+    private final Integer begin;
+    private final Integer end;
+    private final Type type;
+    private final String anchor;
+    private final String uri;
```

Comment:
If it is an URI, then why not use `URI` as type here?

Reply:
I was not sure, whether the java URI class can be used as the xmpp uri. I'll fix that.